wasmtests/select.wat

```diff
@@ -0,0 +1,15 @@
+(module
+  (func $untyped-select (result i32)
+  	i32.const 42
+  	i32.const 24
+  	i32.const 1
+  	select
+  )
+
+  (func $typed-select (result anyref)
+  	ref.null
+  	ref.null
+  	i32.const 1
+  	select (result anyref)
```

Comment:
We should also add a test like
```wat
(func (param anyref) (result anyref)
    ref.null
    local.get 0
    i32.const 1
    select (result anyref))
```

to exercise when the values are distinct subtypes of the typed select's result type. Would be even better if we had enough support for `ref.func` that we could have turn either a `nullref` or a `funcref` into an `anyref` but I don't think we are there yet, are we?

Reply:
Good idea, I'll add that test.

I'm not sure if we can test `funcref` well. Right now the DummyEnvironment used in wasmtests will convert a wasm `ref.func` into a clif `r32/64.null`. So while it would test wasmparser's validation (which has it's own tests), the CL IR would be the same as the other test. (that's if I understand the purpose of wasmtests correctly)